### ISSUE TITLE
`Where-Object -Not` should except `ScriptBlock` (FilterScripts)

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#19483]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Not sure whether this qualifies as a **Bug Report** or should actually be an **Feature Request**.
Anyways, I have selected **Bug Report** as it currently doesn't meet the (my) expectation and there are a lot of other syntaxes that could do the same.

I am coming from [Delete Specific Rows from a CSV File Using PowerShell](https://stackoverflow.com/questions/75980709/delete-specific-rows-from-a-csv-file-using-powershell), and was about to comment to make use of the [**input collection** feature](https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_comparison_operators#common-features):

```PowerShell
$Data =
  [pscustomobject]@{ 'Id' = 0; User = 'Test' },
  [pscustomobject]@{ 'Id' = 1; User = 'Test1' }
```

`-Not <property name>` works as expected:
```PowerShell
$Data |Where-Object -Not Id

Id User
-- ----
 0 Test
```

### Expected behavior

```console
$Data |Where-Object -Not { $_.Id }

Id User
-- ----
 0 Test


```PowerShell
$Data |Where-Object -Not { 'Test1', 'Test2', 'Test3' -eq $_.Id }

Id User
-- ----
 0 Test
```

**Bottom line:**
```PowerShell
'Something' |Where-Object -Not { $True } |Should -BeNullOrEmpty
```
```


### Actual behavior

```console
$Data |Where-Object -Not { $_.Id }

Id User
-- ----
 0 Test
 1 Test1


```PowerShell
$Data |Where-Object -Not { 'Test1', 'Test2', 'Test3' -eq $_.Id }

Id User
-- ----
 0 Test
 1 Test1
```

**Bottom line:**
```PowerShell
'Something' |Where-Object -Not { $True } |Should -BeNullOrEmpty
```

> `InvalidResult: Expected $null or empty, but got Something.`
```


### Error details

_No response_

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.3
PSEdition                      Core
GitCommitId                    7.3.3
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jhoneill** [comment]
I don't think it is a bug, and a think the answer is to put the not in the script block. 

`-not` is a switch and the command line looks like this 

```
Where-Object [-Property] <System.String> [-InputObject <System.Management.Automation.PSObject>] -Not  [<CommonParameters>]
```
So your example is the same as  running  ` $Data | Where-Object -not '$_.id'`

You can run 

```
PS>  $Data | Where-Object -not id

Id User
-- ----
 0 Test
```

"ID" is a string and a valid property name above; using a filter script without any of the script parameters, this does what you want.

```
PS>  $Data | Where-Object { -not $_.Id }

Id User
-- ----
 0 Test
```

**Comment by iRon7** [comment]
@jhoneill,

> it just changes which side of `{` `-not` can go

I agree the benefit might be low for the effort but using the **expression mode** might require an extra pair of parenthesis (probably in rare cases), as in the initial example:

```PowerShell
$Data |Where-Object { -Not ('Test1', 'Test2', 'Test3' -eq $_.User) }
```
vs:
```PowerShell
$Data |Where-Object -Not { 'Test1', 'Test2', 'Test3' -eq $_.User }
```
Which I think is the general benefit of all the `Where-Object` comparison arguments (where `-Not <expression>` is a maverick).

**Comment by jhoneill** [comment]
@iRon7  Yes, it could move the -not AND save a () . It's still a smaller gain than I'd take to the Working group myself.
